karabo_data/geometry2.py

```diff
@@ -346,23 +346,22 @@ def position_all_modules(self, data):
         centre : ndarray
           (y, x) pixel location of the detector centre in this geometry.
         """
-        assert data.shape == (16, 512, 128)
-        size_yx, centre = self._plotting_dimensions()
+        assert data.shape[-3:] == (16, 512, 128)
+        size_yx, centre = self._plotting_dimensions(data.shape)
         out = np.full(size_yx, np.nan, dtype=data.dtype)
-
-        for i, (module, mod_data) in enumerate(zip(self.modules, data)):
-            tiles_data = np.split(mod_data, 8)
-            for j, (tile, tile_data) in enumerate(zip(module, tiles_data)):
-
+        for i, module in enumerate(self.modules):
+            mod_data = data[..., i, :, :]
+            tiles_data = np.split(mod_data, 8, axis=-2)
+            for j, tile in enumerate(module):
+                tile_data = tiles_data[j]
                 # Offset by centre to make all coordinates positive
                 y, x = tile.corner_idx + centre
                 h, w = tile.pixel_dims
-
-                out[y:y+h, x:x+w] = tile.transform(tile_data)
+                out[..., y:y+h, x:x+w] = tile.transform(tile_data)
 
         return out, centre
 
-    def _plotting_dimensions(self):
```

Comment:
I think we should rename this method - plotting implies 2D to me.

Reply:
ok, that makes sense